Surface_mesh_approximation/include/CGAL/Variational_shape_approximation.h

```diff
@@ -1429,13 +1424,17 @@ class Variational_shape_approximation {
         const Point_3 &p0 = m_vpoint_map[source(he, *m_ptm)];
         const Point_3 &p1 = m_vpoint_map[target(he, *m_ptm)];
         const Point_3 &p2 = m_vpoint_map[target(next(he, *m_ptm), *m_ptm)];
-        const FT farea(std::sqrt(CGAL::to_double(CGAL::squared_area(p0, p1, p2))));
-        const Vector_3 fnorm = CGAL::unit_normal(p0, p1, p2);
+        const FT farea = CGAL::approximate_sqrt(CGAL::squared_area(p0, p1, p2));
+        const Vector_3 fnorm =
+          CGAL::collinear(p0, p1, p2) ? CGAL::NULL_VECTOR : CGAL::unit_normal(p0, p1, p2);
```

Comment:
Note that you should use the function Collinear_3 like you are doing with other functors (l141 for example).